controllers/discovery/initDiscoveryUserData.js

```diff
@@ -9,17 +9,23 @@ const {QueryTypes} = require('sequelize');
  * @returns
  */
 const InitDiscoveryUserData = async (req, res) => {
-  let {limit = 50, page = 0} = req.query;
+  let {limit = 50, page = 0, allow_anon_dm} = req.query;
   page = parseInt(page);
 
   const {userId} = req;
 
+  let where_anon_dm = '';
+  if (allow_anon_dm) {
+    allow_anon_dm = allow_anon_dm ? true : false;
+    where_anon_dm = 'AND A.allow_anon_dm = :allow_anon_dm';
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `allow_anon_dm` variable is being reassigned to a boolean value after it's used in the if condition. This could lead to unexpected behavior as the reassignment should happen before the if condition.
```diff
-  let {limit = 50, page = 0, allow_anon_dm} = req.query;
-  page = parseInt(page);
-  
-  const {userId} = req;
-  
-  let where_anon_dm = '';
-  if (allow_anon_dm) {
-    allow_anon_dm = allow_anon_dm ? true : false;
-    where_anon_dm = 'AND A.allow_anon_dm = :allow_anon_dm';
-  }
+  let {limit = 50, page = 0, allow_anon_dm = false} = req.query;
+  page = parseInt(page);
+  
+  const {userId} = req;
+  
+  let where_anon_dm = '';
+  if (allow_anon_dm === 'true') {
+    where_anon_dm = 'AND A.allow_anon_dm = :allow_anon_dm';
+  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->